api/attribute.proto

```diff
@@ -830,10 +830,23 @@ message LsAttributePrefix {
   uint32 sr_prefix_sid = 3;
 }
 
+message LsBgpPeerSegmentSIDFlags {
+  bool value  = 1;
+  bool local = 2;
+  bool backup = 3;
+  bool persistent = 4;
+}
+
+message LsBgpPeerSegmentSID {
+  LsBgpPeerSegmentSIDFlags  flags = 1;
+  uint32 weight  = 2;
+  uint32 sid = 3;
+}
+
 message LsAttributeBgpPeerSegment {
-	uint32 bgp_peer_node_sid = 1;
-	uint32 bgp_peer_adjacency_sid = 2;
-	uint32 bgp_peer_set_sid = 3;
+  LsBgpPeerSegmentSID bgp_peer_node_sid = 1;
+  LsBgpPeerSegmentSID bgp_peer_adjacency_sid = 2;
+  LsBgpPeerSegmentSID bgp_peer_set_sid = 3;
```

Comment:
This breaks the backward compatibility? LsAttributeBgpPeerSegment was introduced in v3.10, right?

Reply:
I just like to understand your intension. I don't ask for workaround. Probably breaking this without the major number updated is fine.

Reply:
Thanks for the review. I'm sorry for not considering compatibility.

As you pointed out, in this PR,  we have changed the gRPC structure to support the flags parameter, which has just been implemented.
This change will enable, for example, the BGP-LS EPE route generation that we plan to do in the future. But on the other hand, compatibility will be affected if existing gRPC clients use this structure.

However, this change has no effect on non-BGP-LS applications. A problem occurs only when using this structure via gRPC. 
Since the gobgp command can now retrieve BGP-LS from this PR, the backward compatibility from the gobgp command in v3.10.0  is not broken.

Thus, in our opinion, it does not break any gRPC operations in existing releases.

If you have any comments, please let us know your opinion.

// FYI, the `message LsAttributePrefix.sr_prefix_sid` should also require the same modification.

Reply:
Sure. Unlikely, this gRPC API change bothers anyone. Pushed, thanks!

> // FYI, the message LsAttributePrefix.sr_prefix_sid should also require the same modification.

like this?

https://github.com/fujita/gobgp/commit/b662da1f77e98cd08accf1946f68a9bdd0c59a71

Reply:
Thank you for your quick response!

> like this?
Sorry, not the same change, but it's a similar kind of change.
The TLVs of each SID in BgpPeeringSegment and the TLV of Prefix-SID are similar in structure, but the meanings of the flags are completely different. In other words, the dedicated SID structure for the Prefix-SID must be prepared.

According to [RFC9085](https://datatracker.ietf.org/doc/html/rfc9085#section-2.3.1), the meaning of flags is different for OSPFv2, OSPFv3, and IS-IS. I think it will be a little more difficult to handle...

Reply:
Better to change the API soon but sounds considerable amount of work...